packages/python/plotly/plotly/express/_imshow.py

```diff
@@ -282,10 +430,30 @@ def imshow(
         layout_patch["margin"] = {"t": 60}
     fig = go.Figure(data=trace, layout=layout)
     fig.update_layout(layout_patch)
-    fig.update_traces(
-        hovertemplate="%s: %%{x}<br>%s: %%{y}<br>%s: %%{z}<extra></extra>"
-        % (labels["x"] or "x", labels["y"] or "y", labels["color"] or "color",)
-    )
+    # Hover name, z or color
+    if binary_string and rescale_image and not (img_rescaled.max() == img.max()):
+        # we rescaled the image, hence z is not displayed in hover since it does
+        # not correspond to img values
```

Comment:
to be discussed. Since we don't have a colorbar maybe it would still be nice to display the z (maybe with another name to make it clear that it's not the original array).

Reply:
on my laptop, computing the max of a (1400, 1400, 3) image array takes 3 ms so it's not a problem to compare the max of `img_rescaled` and `img.max()`.